earthpy/clip.py

```diff
@@ -138,6 +140,45 @@ def clip_shp(shp, clip_obj):
         >>> rmnp_counties = cl.clip_shp(counties, rmnp)
         >>> rmnp_counties.shape
         (4, 13)
+
+    Plotting the clipped sections of the points, lines, and polygons.
+
+    .. plot::
+
+        >>> import matplotlib.pyplot as plt
```

Comment:
heya @mbjoseph  please have a look at this docstring below. It looks like we can only use autodoc plotting with ALL CODE needed to run the plot in the same "chunk". However... i did notice that rasterio and geopandas do seem to handle this differently - ie they don't have to reimport geopandas, etc over and over in their docs.

It seems to me that we don't want super complex docstrings like this and as such maybe we save plots for the more verbose documentation? Thoughts??

Reply:
Heya @lwasser it seems like there are two things here: 

1. Redundant imports for each example. We can solve this by using the [`doctest_global_setup`](https://www.sphinx-doc.org/en/master/usage/extensions/doctest.html#confval-doctest_global_setup) and [`plot_pre_code`](https://matplotlib.org/devel/plot_directive.html#configuration-options) configuration options for Sphinx (basically some edits to `docts/conf.py` to ensure that packages like matplotlib, numpy, and rasterio (and any earthpy modules) are *always* imported for the examples. This is fairly straightforward, and probably worth doing to streamline our docs, especially for imports with consistent aliases like `mpl`, `np`, `gpd`, `rio`, etc. 

2. Very long docstrings. In the example flagged here, even if we remove the imports, the docstring and example is pretty long. If you think it makes sense, we could move these longer examples into vignettes, and restrict examples in docstrings to be **minimal examples** that demonstrate basic functionality.